third-party/realdds/src/dds-device-server.cpp

```diff
@@ -87,13 +88,24 @@ void dds_device_server::publish_image( const std::string & stream_name, const ui
 }
 
 
-static void on_discovery_device_header( size_t const n_streams, dds_notification_server & notifications )
+static void on_discovery_device_header( size_t const n_streams, const dds_options & options,
+                                        dds_notification_server & notifications )
 {
-    topics::flexible_msg notification( json{
+    topics::flexible_msg device_header( json{
         { "id", "device-header" },
         { "n-streams", n_streams },
     } );
-    notifications.add_discovery_notification( std::move( notification ) );
+    notifications.add_discovery_notification( std::move( device_header ) );
+
+    auto opt_json = nlohmann::json();
+    for( auto & opt : options )
+        opt_json.push_back( std::move( opt->to_json() ) );
+    topics::flexible_msg device_options( json {
+        { "id", "device-options" },
+        { "n-options", options.size() },
```

Comment:
Why are we sending the number of streams in the device-header but not the number of options? Why send an options message if there are no options?

Reply:
Number of streams is sent to indicate how many messages to expect. All device options are sent as one message. `n-options` is probably redundant since we can deduce it from array size.

Why not send the options in the device-header message itself?

Reply:
I think it's nice to have one message for options — that can be used for any object, whether device/stream/other. Let's leave as-is for now.